src/TypeAutomata/ToAutomaton.hs

```diff
@@ -26,6 +26,9 @@ import Loc ( defaultLoc )
 import Utils ( enumerate )
 import Control.Monad
 
+--- default Kind where type is not yet clear 
```

Comment:
The occurrence of `defaultKind` is definitely sus ;)

I think you should have enough information in the input to never use the defaultKind here?